sdk/src/main/java/com/deploygate/sdk/DeployGate.java

```diff
@@ -811,10 +963,28 @@ public static void registerCallback(
     }
 
     private void registerCallbackInternal(
-            DeployGateCallback listener,
+            final DeployGateCallback listener,
             boolean callbackImmediately
     ) {
-        mCallbacks.add(listener);
+        mInitializeCallbacks.add(new DeployGateInitializeCallback() {
+            @Override
+            public void onInitialized(boolean isServiceAvailable) {
+                listener.onInitialized(isServiceAvailable);
+            }
+        });
+        mStatusChangeCallbacks.add(new DeployGateStatusChangeCallback() {
+            @Override
+            public void onStatusChanged(boolean isManaged, boolean isAuthorized, String loginUsername, boolean isStopped) {
+                listener.onStatusChanged(isManaged, isAuthorized, loginUsername, isStopped);
+            }
+        });
+        mUpdateAvailableCallbacks.add(new DeployGateUpdateAvailableCallback() {
+            @Override
+            public void onUpdateAvailable(int revision, String versionName, int versionCode) {
+                listener.onUpdateAvailable(revision, versionName, versionCode);
+            }
+        });
```

Comment:
This should be compilable. What happened?

```suggestion
        mInitializeCallbacks.add(listener);
        mStatusChangeCallbacks.add(listener);
        mUpdateAvailableCallbacks.add(listener);
```

The current code breaks unregister methods because their identities have been changed.

Reply:
Sorry.
I added new testcases and checked keeping the behavior between before and after.